felix/nftables/ipsets.go

```diff
@@ -390,48 +390,79 @@ func (s *IPSets) tryResync() error {
 			if err != nil {
 				setsChan <- setData{setName: name, err: err}
 			}
-			strElems := []string{}
-			for _, e := range elems {
-				if len(e.Key) == 3 {
-					// This is a concatination of IP, protocol and port. Format it back into Felix's internal representation.
-					strElems = append(strElems, fmt.Sprintf("%s,%s:%s", e.Key[0], e.Key[1], e.Key[2]))
-				} else {
-					// This is just an IP address / CIDR.
-					strElems = append(strElems, e.Key[0])
-				}
-			}
-			setsChan <- setData{setName: name, elems: strElems}
+			setsChan <- setData{setName: name, elems: elems}
 		}(setName)
 	}
 
 	// We expect a response for every set we asked for.
 	responses := make([]setData, len(sets))
-	for range sets {
+	for i := range sets {
```

Comment:
```suggestion
	for i := range responses {
```
Ranging over the input `sets` hints to me that there's some relationship between `sets[i]` and `responses[i]` but that's not the case; responses will be in random order.